### ISSUE TITLE
False positive for UselessAssignment

### ISSUE THREAD

**Issue by henrik** [r7kamura/slimcop#13]
slimcop 0.17.1

Slim file:

```
- foo = "foo"
= foo
```

Output:

```
$ slimcop /tmp/example.slim
Inspecting 1 file
W

Offenses:

/tmp/example.slim:1:3: W: Lint/UselessAssignment: Useless assignment to variable - foo. (https://rubystyle.guide#underscore-unused-vars)
- foo = "foo"
  ^^^

1 file inspected, 1 offense detected
```

**Comment by r7kamura** [comment]
Good point. I'm going to disable Lint/UselessAssignment by default as well as other similar cops.

**Comment by r7kamura** [comment]
Oh, it's already disabled by default.

https://github.com/r7kamura/slimcop/blob/a2941d4d1fd9836d645a49cc7a9fa5878585cd05/default.yml#L88-L89

@henrik Is this a result of manually enabling Lint/UselessAssignment in your config?

In principle, each piece of Ruby code is treated as an entirely different piece of Ruby code, so Lint/UselessAssignment cannot be used on Slim. This is why we disable it by default.

**Comment by henrik** [comment]
Ah, sorry for not providing enough context about that. And thank you!

Yes indeed, we turn it on in our general Rubocop config for this project. It has caught some real issues as well, so maybe we'll leave it on, just not for autocorrection.

Naive question out of curiosity – what prevents treating all pieces of Ruby code in the same file as though they're all in a single document?

**Comment by r7kamura** [comment]
I see. If there is a cop that you want to apply only to slimcop, I recommend you to create `.slimcop.yml` and it will take precedence over `.ruboocp.yml`. (This should be written in README. I'll add it later)

In my Rails app, for example, I have the following file

```yaml
# .slimcop.yml
inherit_from:
  - .rubocop.yml

Rails/OutputSafety:
  Enabled: false
```

> Naive question out of curiosity – what prevents treating all pieces of Ruby code in the same file as though they're all in a single document?

This is just to simplify the implementation, and in fact it may be possible to achieve this if we try hard enough.

There are about two major issues.

1. One is the location issue: we need to report the exact location of each offense. This becomes more difficult when the code is put together in one file and then inspected.
2. The other issue is that the embedded code is not actually valid Ruby syntax code. For example, you can write `- if a` or `- foo.each` in Slim code, but the `if a` or `foo.each\n  ...` code is not valid Ruby code. In order to handle this kind of code well, Slimcop converts the code a little before passing it to RuboCop, but this sometimes results in a SyntaxError, so it is safer to treat each as separate Ruby code.

**Comment by henrik** [comment]
Thank you so much for explaining! Very helpful.
